sql/core/src/main/scala/org/apache/spark/sql/execution/command/views.scala

```diff
@@ -154,14 +144,11 @@ case class CreateViewCommand(
       } else if (tableMetadata.tableType != CatalogTableType.VIEW) {
         throw new AnalysisException(s"$name is not a view")
       } else if (replace) {
-        // Detect cyclic view reference on CREATE OR REPLACE VIEW.
-        val viewIdent = tableMetadata.identifier
-        checkCyclicViewReference(analyzedPlan, Seq(viewIdent), viewIdent)
```

Comment:
To pass the existing tests, I moved `checkCyclicViewReference` into `rules`. Since the duplication checks also catch the cyclic cases, I think we need to check the cyclic cases first, and then check the name duplication.